node/src/command.rs

```diff
@@ -233,22 +233,30 @@ pub fn run() -> Result<()> {
         },
         #[cfg(feature = "try-runtime")]
         Some(Subcommand::TryRuntime(cmd)) => {
-            let runner = cli.create_runner(cmd)?;
-
+            //Err("try-runtime is deprecated. Use the CLI".into())
+            use crate::common::AvnParachainExecutor;
+            use avn_parachain_runtime::SLOT_DURATION;
             use sc_executor::{sp_wasm_interface::ExtendedHostFunctions, NativeExecutionDispatch};
-            type HostFunctionsOf<E> = ExtendedHostFunctions<
-                sp_io::SubstrateHostFunctions,
-                <E as NativeExecutionDispatch>::ExtendHostFunctions,
-            >;
-
-            // grab the task manager.
-            let registry = &runner.config().prometheus_config.as_ref().map(|cfg| &cfg.registry);
-            let task_manager =
-                sc_service::TaskManager::new(runner.config().tokio_handle.clone(), *registry)
-                    .map_err(|e| format!("Error: {:?}", e))?;
-
-            runner.async_run(|_| {
-                Ok((cmd.run::<Block, HostFunctionsOf<ParachainExecutor>>(), task_manager))
+            use try_runtime_cli::block_building_info::substrate_info;
+
+            let runner = cli.create_runner(cmd)?;
+            runner.async_run(|config| {
+                // we don't need any of the components of new_partial, just a runtime, or a task
+                // manager to do `async_run`.
+                let registry = config.prometheus_config.as_ref().map(|cfg| &cfg.registry);
```

Comment:
you now create the `task_manager` inside the runner? why is this change needed?

Reply:
it doesn't compile with the previous code when you enable try-runtime. This is copied from Substrate